Yolol.Analysis/ControlFlowGraph/Extensions/AnalysisExtensions.cs

```diff
@@ -70,5 +71,38 @@ public static class AnalysisExtensions
 
             return r.Names;
         }
+
+        [NotNull]
+        public static IReadOnlyCollection<VariableName> FindBooleanVariables([NotNull] this IControlFlowGraph cfg, ISingleStaticAssignmentTable ssa)
+        {
+            var booleans = new HashSet<VariableName>();
+            var nonBooleans = new HashSet<VariableName>();
+            var unknowns = new HashSet<VariableName>();
+
+            var count = -1;
+
+            // Keep finding more constants until no more are found
+            while (count != 0)
+            {
+                count = unknowns.Count;
+                cfg.VisitBlocks(() => new FindBooleanVariables(booleans, nonBooleans, unknowns, ssa));
+            }
+
+            var result = new HashSet<VariableName>(booleans.Where(n => !n.IsExternal));
+
+            return result;
+        }
+
+        [NotNull]
+        public static IReadOnlyCollection<Assignment> FindVariableAssignments([NotNull] this IControlFlowGraph cfg, ISingleStaticAssignmentTable ssa)
```

Comment:
It is not clear from the name of this method that it does not find external assignments. I can see three ways to resolve this:
 - Rename to `FindNonExternalVariableAssignments`
 - Make it `private`
 - Add a `bool includeExternals = false` parameter

My preference is `private`

Reply:
I'm going for `includeExternals = false` as this is simple and encourages code reuse.